src/devices/machine/pccard_sram.h

```diff
@@ -0,0 +1,89 @@
+// license: BSD-3-Clause
+// copyright-holders: Dirk Best
+/***************************************************************************
+
+    SRAM PC Cards
+
+***************************************************************************/
+
+#ifndef MAME_MACHINE_PCCARD_SRAM_H
+#define MAME_MACHINE_PCCARD_SRAM_H
+
+#pragma once
+
+#include "machine/pccard.h"
+
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> pccard_sram_device
+
+class pccard_sram_device : public device_t, public device_memory_interface, public device_pccard_interface
+{
```

Comment:
This thing really needs to implement `device_image_interface` and allow the user to supply a file for the content of the card, in the same way that we do for the ATA hard disk device.  If you do it as NVRAM the way you have, you can only have a single memory card of each type per machine per slot.  This is not how these kinds of cards were used – they were used as user storage media for many “appliance” type systems (e.g. sewing and embroidery machines)

Reply:
I'll take a look at this, however it won't be a simple file - the device needs to save the SRAM contents as well the EEPROM contents. Systems can and will change both.

Reply:
You could concatenate the two – use SRAM followed by EEPROM.  You could use some heuristic to determine the division automatically.  There are relatively few non-power-of-two SRAM card sizes (mostly smaller cards in 2 KiB steps up to 16 KiB), and the EEPROM size is always a power-of-two smaller than the page size.

When opening a file, find the least significant set bit of the size.  This gives you the EEPROM size.  After clearing that bit, find the next least significant set bit in the size.  This gives you the page size.

For example if you had a 14 KiB card with a  256-byte EEPROM, the size would be (14 * 1024) + 256 = 14592:
* 14592 = 0x3900 = 0b111001'00000000
* Least significant set bit is bit 8 – EEPROM size = 0b1'00000000 = 0x100 = 256
* After clearing bit 8, size = 0b111000'00000000
* Least significant set bit is bit 11, – page size = 0b1000'00000000 = 0x800 = 2048